community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_3/TransactionBoundPlanContext.scala

```diff
@@ -40,7 +39,7 @@ class TransactionBoundPlanContext(tc: TransactionalContextWrapper)
   extends TransactionBoundTokenContext(tc.statement) with PlanContext with IndexDescriptorCompatibility {
 
   @Deprecated
-  def getIndexRule(labelName: String, propertyKey: String): Option[IndexDescriptor] = evalOrNone {
+  def getIndexRule(labelName: String, propertyKey: String): Option[v2_3.IndexDescriptor] = evalOrNone {
```

Comment:
Rather change NewIndexDescriptor to KernelIndexDescriptor, and remove all v2_3 changes.